contributors/design-proposals/daemonset-create-first-rolling-update.md

```diff
@@ -0,0 +1,177 @@
+## DaemonSet create-first rolling update strategy
+
+DaemonSets currently support a "delete first, then add" rolling update
+strategy.
+
+This document presents a design for a new "add first, then delete" rolling
+update strategy, called `SurgingRollingUpdate`. This strategy will create new
+pods first, cleaning up the old pods once the new ones are running. As with
+Deployments, users will use a `maxSurge` parameter to control how many nodes
```

Comment:
What happens if maxSuge is zero? Would that interrupt the algorithm in this paper? Also what is the default value for maxSuge?